packages/react-components/react-datepicker-compat/src/components/DatePicker/useDatePicker.tsx

```diff
@@ -463,7 +464,18 @@ export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HT
       required: isRequired,
       role: 'combobox',
       tabIndex,
-      ...inputProps,
+      input: {
+        // Needs to be of type any since the union becomes too complex for TS to resolve
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        children: (Component: any, componentProps: any) => (
+          <PopoverTrigger>
+            <Component {...componentProps} />
+          </PopoverTrigger>
```

Comment:
There are some drawbacks to this approach:
* We have to type the component and componentProps as any due to the union being too complex for TS.
* PopoverTrigger seems "hidden" from the render function.

This is the simplest fix to the issue mentioned above.